napari/_qt/qt_main_window.py

```diff
@@ -557,11 +560,29 @@ def closeEvent(self, event):
 
         Regardless of whether cmd Q, cmd W, or the close button is used...
         """
+        try:
+            status = self._plugin_manager_dialog.query_status()
```

Comment:
I thought we wanted the plugin manager to register its status in a global status list in napari, but it looks like we are querying napari-plugin-manager from napari, which means that napari needs to know API details about napari-plugin-manager... Not sure if that's too much coupling. Ideally this bit would check something like `napari.background_tasks` where the plugin-manager has registered its [payloads](https://github.com/napari/napari-plugin-manager/pull/56/files#diff-c4a932d5522a5a713957223899c8991682affd18d838b197d3c508ea32333a1eR1228-R1233).

Reply:
Indeed that was the broader plan but felt like we needed more discussion to do it like that hence why I took this "shortcut" that was meant to be temporary. I can try to open a new PR to check how the other approach could work.